change_note_duration.py

```diff
@@ -0,0 +1,102 @@
+parser.add_argument('-end_delimiter',
+                    default='Verses',
+                    help='A string to search for in the file to indicate where note values end.')
+parser.add_argument('--double', action='store_true', help="Double note duration instead of halving it.")
+
+args = parser.parse_args()
+
+
+class NoteValueError(Exception):
+    pass
+
+
+def get_music_and_lyrics_line_numbers(file_name, start_delimiter, end_delimiter):
+    music_line = None
+    lyrics_line = None
+    with open(file_name, 'r') as file:
+        for (i, line) in enumerate(file):
+            if not music_line and start_delimiter in line:
+                music_line = i
+            if not lyrics_line and end_delimiter in line:
+                lyrics_line = i
+    return (music_line, lyrics_line,)
+
+
+def halve_note_duration(char):
+    try:
+       return str(int(char)*2)
+    except ValueError:
+        return char
+
+def double_note_duration(char):
+    try:
+        note_duration = int(char)
+        if note_duration == 1:
+            raise NoteValueError('Whole notes cannot be doubled')
+        else:
+            return str(note_duration // 2)
+    except ValueError:
+        return char
+
+
+def process_line(line, double_notes):
+    '''Double or halve note duration in a given line
+
+    Splits a string representing notes and note duration into the notes and rhythms,
+    and either halves or doubles those note durations.
+
+    :param line:
+    :param double_notes:
+    :return: new_line
+    '''
+    new_line = ''
+    for char in line:
+        if not double_notes:
+            new_line = "".join(["".join(x) if not is_digit else halve_note_duration("".join(x)) for is_digit, x in
+                                itertools.groupby(line, key=str.isdigit)])
+        else:
+            new_line = "".join(["".join(x) if not is_digit else double_note_duration("".join(x)) for is_digit, x in
+                                itertools.groupby(line, key=str.isdigit)])
+
+    return new_line
+
+
+(music_line, lyrics_line) = get_music_and_lyrics_line_numbers(args.file, args.start_delimiter, args.end_delimiter)
+if not music_line:
+    raise Exception('No start delimiter found. Delimiter: {}'.format(args.start_delimiter))
+if not lyrics_line:
+    raise Exception('No end delimiter found. Delimiter: {}'.format(args.end_delimiter))
+
+temp_file_name = args.file + '~'
+source = open(args.file, 'r')
+dest = open(temp_file_name, 'w')
+try:
+    for i, source_line in enumerate(source):
+            if i > music_line and i < lyrics_line:
+                    line = process_line(source_line, double_notes=args.double)
+            else:
+                line = source_line
+
+            dest.write(line)
+except Exception as exc:
+    source.close()
+    dest.close()
+    os.remove(temp_file_name)
+    raise exc
+
+source.close()
+dest.close()
+os.rename(temp_file_name, args.file)
```

Comment:
What happens if the original file had non-default permissions or group or whatever? The tempfile won't have these so running this could end up changing permissions/group or, like, removing xattrs/acls, or god knows what else.

Reply:
I don't think that's a common use case here, but maybe something to think about.

Reply:
Issue #17